src/metrics.jl

```diff
@@ -141,6 +163,28 @@ chisq_dist(a::AbstractArray, b::AbstractArray) = evaluate(ChiSqDist(), a, b)
 @inline eval_reduce(::KLDivergence, s1, s2) = s1 + s2
 kl_divergence(a::AbstractArray, b::AbstractArray) = evaluate(KLDivergence(), a, b)
 
+# RenyiDivergence
+function eval_start{T<:AbstractFloat}(::RenyiDivergence, a::AbstractArray{T}, b::AbstractArray{T})
+    zero(T), zero(T)
+end
+
+@inline eval_op(dist::RenyiDivergence, ai, bi) = ai, (ai == 0 ? zero(ai) :
+                                                      (dist.is_normal ?
+                                                       ai .* ((ai ./ bi) .^ dist.p) :
+                                                       (dist.is_zero ?
+                                                        bi :
+                                                        (dist.is_one ?
+                                                         ai * log(ai / bi) :
+                                                         # otherwise α = ∞
+                                                         ai / bi))))
```

Comment:
This is pretty hard to read. Would you mind putting it into the long form function syntax with `if` statements in place of ternaries? E.g.
```julia
@inline function eval_op(dist::RenyiDivergence, ai, bi)
    if ai == 0
        zero(ai)
    ...
end
```